collections/soldiers.js

```diff
@@ -26,4 +26,36 @@ function newSoldier(data, callback) {
   }
 }
 
-module.exports.newSoldier = newSoldier;
\ No newline at end of file
+function getSoldier(name, id, callback) {
+  MongoClient.connect(url, function (err, client) {
```

Comment:
MongoClient.connect  is a very heavy async function, it should only ever be done once in initialization and never again